src/pveclib/vec_f128_ppc.h

```diff
@@ -4957,7 +6260,287 @@ static inline vec_xscvudqp (vui64_t int64)
+      b_sign = vec_setb_sq (int128);
+      q_sig = vec_seluq ((vui128_t) int128, q_neg, b_sign);
+      // Start with the quad exponent bias + 127 then subtract the count of
+      // leading '0's. The 128-bit sig can have 0-127 leading '0's.
+      vui64_t q_expm = (vui64_t) CONST_VINT64_DW(0, (0x3fff + 127));
+      vui64_t i64_clz = (vui64_t) vec_clzq (q_sig);
+      q_sig = vec_slq (q_sig, (vui128_t) i64_clz);
+      q_exp = vec_subudm (q_expm, i64_clz);
+      // This is the part that might require rounding.
+
+      // The Significand (including the L-bit) is right justified in
+      // in the high-order 113-bits of q_sig.
+      // The guard, round, and sticky (GRX) bits are in the low-order
+      // 15 bits.
+      // The sticky-bits are the last 13 bits and are logically ORed
+      // (or added to 0x1fff) to produce the X-bit.
+      //
+      // For "round to Nearest, ties to even".
+      // GRX = 0b001 - 0b011; truncate
+      // GRX = 0b100 and bit-112 is odd; round up, otherwise truncate
+      // GRX = 0b100 - 0b111; round up
+      // We can simplify by copying bit-112 and OR it with bit-X
+      // Then add 0x3fff to q_sig will generate a carry into bit-112
+      // if and only if GRX > 0b100 or (GRX == 0b100) && (bit-112 == 1)
+      const vui32_t RXmask = CONST_VINT128_W( 0, 0, 0, 0x3fff);
+      vui128_t q_carry, q_sigc;
+      vb128_t qcmask;
+      vui32_t q_odd;
+      // Isolate bit-112 and OR into GRX bits if q_sig is odd
+      q_odd = (vui32_t) vec_srhi ((vui16_t)q_sig, 15);
+      q_odd = vec_and (q_odd, lowmask);
+      q_sig = (vui128_t) vec_or ((vui32_t) q_sig, q_odd);
+      // We add 0x3fff to GRX-bits which may carry into low order sig-bit
+      // This may result in a carry out of bit L into bit-C.
+      q_carry = vec_addcuq (q_sig, (vui128_t) RXmask);
+      q_sig = vec_adduqm (q_sig, (vui128_t) RXmask);
+      // Generate a bool mask from the carry to use in the vsel
+      qcmask = vec_setb_cyq (q_carry);
+      // Two cases; 1) We did carry so shift (double) left 112 bits
+      q_sigc = vec_sldqi (q_carry, q_sig, 112);
+      // 2) no carry so shift left 15 bits
+      q_sig = vec_srqi ((vui128_t) q_sig, 15);
+      // Select which based on carry
+      q_sig = (vui128_t) vec_sel ((vui32_t) q_sig, (vui32_t) q_sigc, (vui32_t) qcmask);
+      // Increment the exponent based on the carry
+      q_exp = vec_addudm (q_exp, (vui64_t) q_carry);
+
+      q_exp = vec_swapd (q_exp);
+      // Copy Sign-bit to QP significand before insert.
+      q_sig = (vui128_t) vec_or ((vui32_t) q_sig, q_sign);
+      result = vec_xsiexpqp (q_sig, q_exp);
+    }
+#else
+  result = int128[0];
+#endif
+  return result;
+}
+
+/** \brief VXS Scalar Convert Unsigned-Quadword to Quad-Precision format.
+ *
+ *  The unsigned quadword element of vector int128 is converted
+ *  to quad-precision format.
+ *  If the conversion is not exact the default rounding mode is
+ *  "round to Nearest, ties to even".
```

Comment:
s/"round to Nearest, ties to even"/Round to Even/

Reply:
"Round to Nearest Even"